### PR TITLE
NFDIV-4416

### PR DESCRIPTION
**Pull request by DelythJustice**
### Change description ###

Enter a description.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-4416

### Pull request checklist ###

**Before raising**
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**Before merging**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

**Note:** Bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -428,7 +425,7 @@ bootWithCCD {
     environment 'CITIZEN_UPDATE_CASE_STATE_ENABLED', 'true'
     environment 'LOGGING_LEVEL_UK_GOV_HMCTS_DIVORCE', 'info'
     environment 'ROLE_ASSIGNMENT_URL', 'http://localhost:4096'
-    environment 'CASE_DATA_STORE_BASEURL', 'http://localhost:4452'
+ //   environment 'CASE_DATA_STORE_BASEURL', 'http://localhost:4452'
```
Comment (adamg-hmcts) [inline build.gradle:428]: Are you planning to leave in this change?

File: src/cftlib/java/uk/gov/hmcts/divorce/cftlib/CftLibConfig.java [hunk 2]
```diff
@@ -81,5 +86,9 @@ public void configure(CFTLib lib) throws Exception {
         // Import CCD definitions
         lib.importJsonDefinition(new File("build/definitions/NFD"));
         lib.importJsonDefinition(new File("build/definitions/NO_FAULT_DIVORCE_BulkAction"));
+
+        Path filePath = Paths.get("resources/ccd-DIVORCE.xlsx");
```
Comment (adamg-hmcts) [inline src/cftlib/java/uk/gov/hmcts/divorce/cftlib/CftLibConfig.java:90]: Nitpick: Do you think we should rename it to `ccd-OLD-DIVORCE.xlsx` to help distinguish it from our no fault divorce CCD definitions?

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java [hunk 3]
```diff
@@ -160,4 +204,28 @@ public void setToNewMatches(CaseData data, List<CaseMatch> newMatches) {
             data.setCaseMatches(null);
         }
     }
+
+    List<uk.gov.hmcts.reform.ccd.client.model.CaseDetails> getOldDivorceFreshMatches(MarriageDetails marriageDetails) {
+        BoolQueryBuilder nameMatchQuery1 = QueryBuilders.boolQuery()
+            .filter(QueryBuilders.termQuery("data.D8MarriageRespondentName.keyword", marriageDetails.getApplicant1Name()))
+            .filter(QueryBuilders.termQuery("data.D8MarriagePetitionerName.keyword", marriageDetails.getApplicant2Name()));
```
Comment (adamg-hmcts) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java:211]: Do you know whether the name matching should be case-insensitive?

File: src/main/java/uk/gov/hmcts/divorce/idam/IdamService.java [hunk 4]
```diff
@@ -13,13 +15,20 @@
...
+@Slf4j
 public class IdamService {
     @Value("${idam.systemupdate.username}")
     private String systemUpdateUserName;
 
     @Value("${idam.systemupdate.password}")
     private String systemUpdatePassword;
 
+    @Value("${idam.divorce.username}")
+    private String divorceUserName;
```
Comment (adamg-hmcts) [inline src/main/java/uk/gov/hmcts/divorce/idam/IdamService.java:27]: Should we rename to oldDivorce (same suggestion as the CCD def naming)?

File: build.gradle (hunk 1 above)
Comment (DelythJustice) [reply build.gradle:428]: oops, no, ta

File: src/cftlib/java/uk/gov/hmcts/divorce/cftlib/CftLibConfig.java (hunk 2 above)
Comment (DelythJustice) [reply src/cftlib/java/uk/gov/hmcts/divorce/cftlib/CftLibConfig.java:90]: this reminds me I need to make sure these get plucked out not just in dev env
Can rename, yeh

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java (hunk 3 above)
Comment (DelythJustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java:211]: it shouldn't be

Comment (DelythJustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java:211]: I'll change to matchQuery

Comment (DelythJustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java:211]: actually, no, cos it should be exact match, I'll think of something else

Comment (DelythJustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java:211]: hm, we have lowercase normalizer on our elastic, I'll need to check the one for old Divorce has it to be sure

File: src/cftlib/java/uk/gov/hmcts/divorce/cftlib/CftLibConfig.java (hunk 2 above)
Comment (DelythJustice) [reply src/cftlib/java/uk/gov/hmcts/divorce/cftlib/CftLibConfig.java:90]: renamed

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java (hunk 3 above)
Comment (DelythJustice) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java:211]: I meant it should be case-insensitive, I misread your comment.
I checked in AAT, it picks up a case that's got caps at start of name in nfd and matches with a case that's lower case for whole name in old divorce, so it must be same lowercase es mapping

**Review by adamg-hmcts — DISMISSED** [review]

**Review by adamg-hmcts — APPROVED** [review]

**Pull request merged**